src/main/webapp/app/oncokb-annotation-visualisation/AnnotationVisualisation.tsx

```diff
@@ -0,0 +1,380 @@
+    const savedAnnotationColumns = localStorage.getItem(
+      'selectedAnnotationColumns'
+    );
+    const savedTreatmentColumns = localStorage.getItem(
+      'selectedTreatmentColumns'
+    );
+
+    if (savedAnnotationColumns && savedTreatmentColumns) {
+      this.setState({
+        selectedAnnotationColumns: JSON.parse(savedAnnotationColumns),
+        selectedTreatmentColumns: JSON.parse(savedTreatmentColumns),
+      });
+    }
+  }
+
+  handleAnnotationColumnsChange = (selectedOptions: any) => {
+    this.setState({
+      selectedAnnotationColumns: selectedOptions.map((option: any) =>
+        option.value ? option.value : option.prop
+      ),
+    });
+    localStorage.setItem(
+      'selectedAnnotationColumns',
+      JSON.stringify(selectedOptions.map((option: any) => option.value))
+    );
+  };
+
+  handleTreatmentColumnsChange = (selectedOptions: any) => {
+    this.setState({
+      selectedTreatmentColumns: selectedOptions.map(
+        (option: any) => option.value
+      ),
+    });
+    localStorage.setItem(
+      'selectedTreatmentColumns',
+      JSON.stringify(selectedOptions.map((option: any) => option.value))
+    );
+  };
+
+  @computed
+  get treatmentTableColumns(): SearchColumn<TreatmentImplication>[] {
+    const selectedKeys = this.state.selectedTreatmentColumns;
+    const filteredColumns = treatmentColumns.filter(col =>
+      selectedKeys.includes(col.key)
+    );
+    const hasAnnotation = selectedKeys.some(
+      column => column === TREATMENTS_TABLE_COLUMN_KEY.ANNOTATION
+    );
+    const totalColumns = selectedKeys.length;
+    const calculatedWidth = hasAnnotation
+      ? 800 / (totalColumns - 1)
+      : 1200 / totalColumns;
+    return filteredColumns.map(column => ({
+      ...getDefaultColumnDefinition(column.key, calculatedWidth),
+      onFilter: (data: TreatmentImplication, keyword) =>
+        filterByKeyword(data[column.prop], keyword),
+    }));
+  }
+
+  @computed
+  get annotationTableColumns(): SearchColumn<AnnotationImplication>[] {
+    const selectedKeys = this.state.selectedAnnotationColumns;
+    const filteredColumns = annotationColumns.filter(col =>
+      selectedKeys.includes(col.key)
+    );
+
+    return filteredColumns.map(column => ({
+      ...getDefaultColumnDefinition(
+        column.key,
+        1200 / this.state.selectedAnnotationColumns.length
+      ),
+      onFilter: (data: AnnotationImplication, keyword) =>
+        filterByKeyword(data[column.prop], keyword),
+    }));
+  }
+
+  filterAnnotationsByType(alterationType: string): AnnotationImplication[] {
+    return this.props.annotations.filter(
+      annotation => annotation.alterationType === alterationType
+    );
+  }
+
+  filterTreatmentsByType(alterationType: string): TreatmentImplication[] {
+    return this.props.treatments.filter(
+      treatment => treatment.alterationType === alterationType
+    );
+  }
+
+  render() {
+    return (
+      <>
+        {this.props.isPatientInfoVisible && (
```

Comment:
It looks like `patientId` and `patientInfo` are only used when `isPatientInfoVisible === true`. However, the first two props I mentioned are not optional. I would prefer to have these bundled into an optional object like the following:

```
patientInfo?: {
    id: string
    info: string
}
```

`id` and `info` could also be optional.